src/main/java/sirius/kernel/commons/Outcall.java

```diff
@@ -192,11 +194,39 @@ public Outcall(URI uri) {
      * Allows to modify the client before the request is sent by returning the builder that is used to create it.
      *
      * @return the underlying {@link HttpClient.Builder}
+     * @deprecated Use {@link #modifyClient(String)}
      */
+    @Deprecated
```

Comment:
Whats the rationale behind deprecating the methods without the `clientSelector`? Often we only need one request, and with simple ones like
```java
Outcall call = new Outcall(URI.create(SERVER_URL));
String response = call.postData(params, StandardCharsets.ISO_8859_1).getData();
```
no `clientSelector` needs to be specified. If however we want to add a cookie manager or want to trust self-signed certificates, suddenly we need to specify `null`. Either it should be always mandatory, so maybe we should add the `clientSelector` to the constructor, or it should be optional, maybe even with a new `withClientSelector` method.